nuggylib/minecraft-asset-reader
# Simplify CLI operations

# Description

This PR makes the overall application (including the webapp) a bit more "sensical" to use. Before, the CLI was intended to be used for much of the operations that we wanted to support (such as exporting data to a CMS provider). However, I've since realized it will be quite a bit more usable if we have the majority of the user interface in the webapp since this tool is geared towards helping newer developers (e.g., people who may not be fully comfortable with a CLI-exclusive interface).

Main changes introduced in this PR:
1. No more `SiteData`; this was a type that was going to be used to hold an "in-memory" reference to the site data to generate, but the information can be inferred by looking at the `ContentMap` - there is no need to store the site data in-memory
2. Generate site data for blocks (including JSON files and images)
3. Nest exported data into a timestamp folder at the specified export path


## Testing steps
1. Run `yarn start` > specify the assets path to import from (point it to a valid minecraft assets directory, like the `assets` dir in `1.12.2`)
2. Run the webapp ([separate repo](https://github.com/nuggylib/minecraft-asset-reader-webapp)) 
3. Configure some blocks 
4. Open the side menu > Click export > configure export settings (make sure path is valid)
5. Export
6. Confirm that the expected content was exported/generated (there should be both JSON files and actual image files of the block icons for the various specified sizes)
